test/unit/RepetitionTaskTest.js

```diff
@@ -23,45 +23,41 @@ describe('RepetitionTaskTest', function () {
 
     it('should be cancelled before timeout', async function () {
         let counter = 0;
-        const task = Util.scheduleWithRepetition(function () {
-            counter++;
-        }, 50, 75);
+        const task = Util.scheduleWithRepetition(() => counter++, 50, 100);
 
-        await TestUtil.promiseWaitMilliseconds(40);
+        await TestUtil.promiseWaitMilliseconds(25);
         Util.cancelRepetitionTask(task);
         expect(counter).to.be.equal(0);
-        await TestUtil.promiseWaitMilliseconds(130);
+        await TestUtil.promiseWaitMilliseconds(150);
         expect(counter).to.be.equal(0);
     });
 
     it('should be cancelled after timeout', async function () {
         let counter = 0;
-        const task = Util.scheduleWithRepetition(function () {
+        const task = Util.scheduleWithRepetition(() => {
             counter++;
-        }, 50, 75);
+        }, 50, 100);
 
-        await TestUtil.promiseWaitMilliseconds(60);
+        await TestUtil.promiseWaitMilliseconds(100);
         Util.cancelRepetitionTask(task);
         expect(counter).to.be.equal(1);
-        await TestUtil.promiseWaitMilliseconds(75);
+        await TestUtil.promiseWaitMilliseconds(100);
         expect(counter).to.be.equal(1);
     });
 
     it('should be cancelled after interval', async function () {
         let counter = 0;
-        const task = Util.scheduleWithRepetition(function () {
-            counter++;
-        }, 50, 75);
+        const task = Util.scheduleWithRepetition(() => counter++, 50, 100);
 
-        await TestUtil.promiseWaitMilliseconds(130);
+        await TestUtil.promiseWaitMilliseconds(200);
         Util.cancelRepetitionTask(task);
         expect(counter).to.be.equal(2);
-        await TestUtil.promiseWaitMilliseconds(75);
+        await TestUtil.promiseWaitMilliseconds(50);
```

Comment:
Minor: Should this be something larger than 50 ms(maybe 75ms) to make sure that the task is not run the third time? Because the total wait time is 250ms and the task is supposed to run three times after 250ms of its creation, so I am not sure whether or not this test would deterministically fail if there were some problems in the implementation

Reply:
Good suggestion. Changed the wait time to 75ms. See 6fdda4049efb1311921d549ed83840adfdba8eb9